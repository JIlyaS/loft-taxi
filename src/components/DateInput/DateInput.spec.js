import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { MuiPickersUtilsProvider } from '@material-ui/pickers';
import DateFnsUtils from '@date-io/date-fns';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { render, screen, fireEvent } from '@testing-library/react';

import {Provider} from 'react-redux';
import createStore from '../../store';
import {DateInput} from './';

const store = createStore();

const defaultProps = {
  id: 'date-input',
  label: '',
  value: new Date(),
  classNameWrap: '',
  isRequired: false,
  setSelectedDate: () => {}
};

const TestDateInput = props => <DateInput {...defaultProps} {...props} />;

describe('DateInput', () => {
  it('renders correctly snapshot', () => {
    const history = createMemoryHistory();
    const tree = renderer.create(
      <Provider store={store}>
        <Router history={history}>
          <MuiPickersUtilsProvider utils={DateFnsUtils}>
            <TestDateInput />
          </MuiPickersUtilsProvider>
        </Router>
      </Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
    // it("render card correctly with null props", () => {
    //   const props = {
    //     id: 'date-input',
    //     label: '',
    //     value: new Date(),
    //     classNameWrap: '',
    //     isRequired: false,
    //     setSelectedDate: () => {}
    //   };
    //   const DateInputComponent = mount(<MuiPickersUtilsProvider utils={DateFnsUtils}>
    //     <TestDateInput {...props} />
    //   </MuiPickersUtilsProvider>);
    //   expect(DateInputComponent.prop("value")).toEqual(new Date());
    //   expect(DateInputComponent.prop("id")).toEqual('date-input');
    // });
});