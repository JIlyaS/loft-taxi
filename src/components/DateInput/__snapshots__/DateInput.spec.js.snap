// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateInput renders correctly snapshot 1`] = `
<div
  className="input"
>
  <div
    className="MuiFormControl-root MuiTextField-root PickerWithState-root-1 MuiFormControl-marginDense"
  >
    
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd MuiInputBase-marginDense MuiInput-marginDense"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
        disabled={false}
        id="date-input"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder=""
        required={false}
        type="text"
        value="06/2021"
      />
      <div
        className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-marginDense"
      >
        <button
          aria-label="change date"
          className="MuiButtonBase-root MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
    
  </div>
</div>
`;
